src/common/reducers/entities/index.js

```diff
@@ -0,0 +1,27 @@
+import merge from 'lodash/merge';
+
+import * as RepoActionTypes from '../../actions/repository';
+
+import repository from './repository';
+
+
+// TODO snaps
+
+export function entities(state = { snaps: {}, repos: {} }, action) {
+  if (action.payload && action.payload.entities) {
+    return merge({}, state, action.payload.entities);
+  }
+
+  // only modify repos if action is one of REPO_ types
+  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
+    return {
+      ...state,
+      repos: {
```

Comment:
I've flip-flopped in my opinion on this a few times during this work. FWIW I don't agree that the logic here is weird and better use of reducers means components will be looking at this directly less, but I agree that we'll find out more when we start on snaps.